core/user_controller/UserController.py

```diff
@@ -1,6 +1,178 @@
+from ta_scheduler.models import User, Course, CourseSection, LabSection, TALabAssignment, TACourseAssignment
+from django.core.exceptions import ValidationError, PermissionDenied, ObjectDoesNotExist
+from django.db import models
+from django.shortcuts import get_object_or_404
 
 
+class UserController:
 
+    @staticmethod
+    def getUser(user_id):
+        """
+        Preconditions: user_id is a valid id matching a record of a user in the Users table.
+        Postconditions: Retrieves the user with id of user_id, along with a list of the courses associated with the user.
+        If the user is a TA then any assignments are also returned embedded into their courses,
+        if the user is an instructor/administrator then any sections they teach are returned embedded into their courses.
+        Side-effects: none.
+        Parameters: user_id: an id matching a user_id stored in the Users table.
+        Returns: Dictionary containing user, courses, course_sections, lab_sections, lab_assignments, and course_assignments linked to the specified user.
+        """
 
-class UserController:
-    pass # TODO: Create static controller class
\ No newline at end of file
+        if not isinstance(user_id, int) or not user_id:
+            raise ValueError("Invalid user_id: must be a non-empty integer")
+
+        user = get_object_or_404(User, id=user_id)
+
+        courses = Course.objects.filter(
+            models.Q(coursesection__instructor=user) |
+            models.Q(tacourseassignment__ta=user) |
+            models.Q(labsection__talabassignment__ta=user)
+        ).distinct()
+
+        course_sections = CourseSection.objects.filter(instructor=user)
+        lab_assignments = TALabAssignment.objects.filter(ta=user).select_related('lab_section')
+        course_assignments = TACourseAssignment.objects.filter(ta=user)
+
+        lab_sections = LabSection.objects.filter(
+            models.Q(course__in=courses) |
+            models.Q(id__in=lab_assignments.values_list('lab_section__id', flat=True))
+        ).distinct()
+
+        result = {
+            'user': user,
```

Comment:
can we refactor this to use the local_data_classes? I think this would be the UserProfile/PrivateUserProfile and it has some embedded data classes.